queries.sql

```diff
@@ -103,4 +103,57 @@ WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
 
 -- Who owns the most animals?
 SELECT owners.full_name, COUNT(animals.name) AS max_number FROM owners
-JOIN animals ON owners.id = animals.owner_id GROUP BY owners.full_name ORDER BY max_number DESC;
\ No newline at end of file
+JOIN animals ON owners.id = animals.owner_id GROUP BY owners.full_name ORDER BY max_number DESC;
+
+-- Phase 4 add "join table" for visits
+
+-- Who was the last animal seen by William Tatcher?
+SELECT vet.name AS vet_name, an.name AS animal_name
+  FROM visits v JOIN animals an ON an.id = v.animal_id
+  JOIN vets vet ON vet.id = v.vet_id
+  WHERE vet.name = 'William Tatcher'
+  ORDER BY v.date_of_visit DESC LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT vet.name as vet_name, COUNT(date_of_visit) 
+  FROM visits v JOIN vets vet ON v.vet_id = vet.id
+  WHERE name = 'Stephanie Mendez' GROUP BY vet.name;
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT vets.name, species.name FROM vets JOIN specializations ON vets.id = specializations.vet_id JOIN species ON species.id = specializations.species_id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT an.name, vet.name AS vet_name, v.date_of_visit
+  FROM visits v JOIN animals an ON an.id = v.animal_id
+  JOIN vets vet ON vet.id = v.vet_id
+  WHERE vet.name = 'Stephanie Mendez' AND v.date_of_visit 
+    BETWEEN 'Apr 1, 2020' AND 'Aug 30, 2020';
+
+-- What animal has the most visits to vets?
+SELECT an.name, COUNT(an.name) AS most_visits
+  FROM visits v JOIN animals an ON an.id = v.animal_id
+  GROUP BY an.name ORDER BY most_visits DESC LIMIT 1;
+
+-- Who was Maisy Smith's first visit?
+SELECT an.name AS animal, vet.name AS vet, v.date_of_visit
+  FROM visits v JOIN animals an ON an.id = v.animal_id
+  JOIN vets vet ON vet.id = v.vet_id
+  WHERE vet.name = 'Maisy Smith' ORDER BY v.date_of_visit ASC LIMIT 1;
+
+  -- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT an.id AS animal_id, an.name AS animal, an.date_of_birth, vet.id AS vet_id, vet.name AS vet, vet.age AS vet_age, date_of_visit
+  FROM visits v JOIN animals an ON an.id = v.animal_id
+  JOIN vets vet ON vet.id = v.vet_id;
```

Comment:
- Kindly write a query to get the Details for `MOST RECENT` visit: animal information, vet information, and date of visit.

This 👇 is what I'm currently getting when I run your query

![](https://i.imgur.com/Ghupw8t.png)

You can add the following to your existing query 😉 

```
ORDER BY v.date_of_visit DESC
  LIMIT 1;
```

Reply:
Thanks for your review. The change has been implemented.